src/Query/Builder.php

```diff
@@ -1057,41 +1061,39 @@ protected function compileWhereBasic(array $where): array
     {
         extract($where);
 
-        // Replace like or not like with a Regex instance.
+        // Replace like with a Regex instance.
         if (in_array($operator, ['like', 'not like'])) {
-            if ($operator === 'not like') {
-                $operator = 'not';
-            } else {
-                $operator = '=';
-            }
-
-            // Convert to regular expression.
-            $regex = preg_replace('#(^|[^\\\])%#', '$1.*', preg_quote($value));
-
-            // Convert like to regular expression.
-            if (! Str::startsWith($value, '%')) {
-                $regex = '^'.$regex;
-            }
-            if (! Str::endsWith($value, '%')) {
-                $regex .= '$';
-            }
+            // Convert % and _ to regex, and unescape \% and \_
+            $regex = preg_replace(
+                ['#(^|[^\\\])%#', '#(^|[^\\\])_#', '#\\\\\\\(%|_)#'],
+                ['$1.*', '$1.', '$1'],
+                preg_quote($value),
+            );
```

Comment:
I noted there is one test below for escaped `%` and `_` operators: `'a\%cm\_e'`. Thinking aloud to make sense of this:

In the first two replacements, `(^|[^\\\])` is necessary to match either the start of a string or a character that _isn't_ a backslash, and leave that prefix in place when the `%` or `_` is replaced with a `.*` or `.`, respectively. Extra escaping is required because of the following `preg_quote($value)`.

In the final replacement, if we ignore the PHP string escapes the actual pattern is `\\\\(%|_)`. That translates to an escaped backslash in the output of `preg_quote($value)`, which means a literal backslash in the original string. Therefore: `\%` and `\_` in the original string become `%` and `_`, respectively.

That seems reasonable.

Reply:
I added comments to explain the regexes.
They have been updated to match consecutive `%%` or `__`.